sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -181,6 +181,19 @@ public virtual EventHubProperties CreateEventHubPropertiesFromResponse(AmqpMessa
                 throw new InvalidOperationException(string.Format(CultureInfo.CurrentCulture, Resources.InvalidMessageBody, typeof(AmqpMap).Name));
             }
 
+            object geoReplicationFactor = responseData[AmqpManagement.ResponseMap.GeoReplicationFactor];
+            var geoReplicationFactorValue = geoReplicationFactor as int?;
+            if (geoReplicationFactorValue != null)
+            {
+                return new EventHubProperties((string)responseData[AmqpManagement.ResponseMap.Name],
+                                              new DateTimeOffset((DateTime)responseData[AmqpManagement.ResponseMap.CreatedAt], TimeSpan.Zero),
+                                              (string[])responseData[AmqpManagement.ResponseMap.PartitionIdentifiers],
+                                              geoReplicationFactorValue > 1);
+            }
+
+            // This is only because this is a preview feature and the service might not return the geo replication factor.
+            // This should be removed once the geo replication factor is publicly available.
+
             return new EventHubProperties(
                 (string)responseData[AmqpManagement.ResponseMap.Name],
                 new DateTimeOffset((DateTime)responseData[AmqpManagement.ResponseMap.CreatedAt], TimeSpan.Zero),
```

Comment:
```suggestion
            var geoReplicationEnabled = responseData[AmqpManagement.ResponseMap.GeoReplicationFactor] switch
            {
                int count when count > 1 => true,
                _ => false
            };
            
            return new EventHubProperties(
                (string)responseData[AmqpManagement.ResponseMap.Name],
                new DateTimeOffset((DateTime)responseData[AmqpManagement.ResponseMap.CreatedAt], TimeSpan.Zero),
                (string[])responseData[AmqpManagement.ResponseMap.PartitionIdentifiers],
                geoReplicationEnabled);
```

We don't need to a `ContainsKey` check, because the [indexer](https://github.com/Azure/azure-amqp/blob/943835ed5e085b3c06da7a8020a2585254d0e6e1/src/Encoding/AmqpMap.cs#L51) of AmqpMap does a TryGet and returns `null` if the key is not present.

Reply:
Yeah, I noticed that about the indexer, the null check is because the cast to int gives an exception if the value is null, so using `as` instead returns null if the cast doesn't work (i.e. the value doesn't exist).

Reply:
Pattern matching removes the need for the null check.  If it can't be safely cast to an `int`, it'll just fall into the catch-all.

Reply:
Oohh ok that's nice